sql/rowexec: TestDrainingProcessorSwallowsUncertaintyError failed

[(sql/rowexec).TestDrainingProcessorSwallowsUncertaintyError failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1919982&tab=buildLog) on [master@b8ecf62deb88da45a344912f721b0609b5df2187](https://github.com/cockroachdb/cockroach/commits/b8ecf62deb88da45a344912f721b0609b5df2187):

```
I200504 20:52:02.554308 14188 kv/kvserver/stores.go:255  [n3] wrote 2 node addresses to persistent storage
I200504 20:52:02.558119 14191 kv/kvserver/stores.go:255  [n1] wrote 2 node addresses to persistent storage
I200504 20:52:02.561026 14684 kv/kvserver/stores.go:255  [n2] wrote 2 node addresses to persistent storage
I200504 20:52:02.590011 14188 server/node.go:572  [n3] bootstrapped store [n3,s3]
I200504 20:52:02.591145 14188 server/node.go:472  [n3] node=3: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200504 20:52:02.592613 14188 server/server.go:1435  [n3] starting https server at 127.0.0.1:37737 (use: 127.0.0.1:37737)
I200504 20:52:02.592761 14188 server/server.go:1440  [n3] starting postgres server at 127.0.0.1:36471 (use: 127.0.0.1:36471)
I200504 20:52:02.595772 14188 server/server.go:1442  [n3] starting grpc server at 127.0.0.1:34289
I200504 20:52:02.595877 14188 server/server.go:1443  [n3] advertising CockroachDB node at 127.0.0.1:34289
I200504 20:52:02.593565 15644 server/server.go:1066  active cluster version is now 20.1-2 (up from 19.1-1)
I200504 20:52:02.742376 15707 sql/temporary_schema.go:453  [n3] skipping temporary object cleanup run as it is not the leaseholder
I200504 20:52:02.742630 15707 sql/temporary_schema.go:454  [n3] completed temporary object cleanup job
I200504 20:52:02.742999 15707 sql/temporary_schema.go:581  [n3] temporary object cleaner next scheduled to run at 2020-05-04 21:22:02.739480928 +0000 UTC
I200504 20:52:02.768426 14188 server/server_sql.go:636  [n3] done ensuring all necessary migrations have run
I200504 20:52:02.768613 14188 server/server.go:1753  [n3] serving sql connections
I200504 20:52:02.774359 15721 sqlmigrations/migrations.go:663  [n3] starting wait for upgrade finalization before schema change job migration
I200504 20:52:02.784960 15721 sqlmigrations/migrations.go:689  [n3] schema change job migration already complete
I200504 20:52:02.832404 15706 sql/event_log.go:132  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:127.0.0.1:34289 Attrs: Locality:region=test,dc=dc3 ServerVersion:20.1-2 BuildTag:v20.1.0-beta.4-1287-gb8ecf62deb StartedAt:1588625522552483286 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:36471} ClusterID:7e5f4dce-7168-4c3b-9ad2-f34bf179c058 StartedAt:1588625522552483286 LastUp:1588625522552483286}
I200504 20:52:02.852644 15705 server/server_update.go:55  [n3] no need to upgrade, cluster already at the newest version
I200504 20:52:02.853458 14247 gossip/gossip.go:1539  [n1] node has connected to cluster via gossip
I200504 20:52:02.854166 14247 kv/kvserver/stores.go:255  [n1] wrote 2 node addresses to persistent storage
I200504 20:52:02.865997 15684 sql/event_log.go:132  [n1,client=127.0.0.1:47614,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.range_merge.queue_enabled Value:false User:root}
I200504 20:52:03.003232 15684 sql/event_log.go:132  [n1,client=127.0.0.1:47614,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I200504 20:52:03.057381 15684 sql/event_log.go:132  [n1,client=127.0.0.1:47614,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:test.public.t Statement:CREATE TABLE test.public.t (x INT8 PRIMARY KEY) User:root}
I200504 20:52:03.186953 15684 kv/kvserver/replica_command.go:397  [n1,s1,r32/1:/{Table/36-Max}] initiating a split of this range at key /Table/53/1/6 [r33] (manual)
I200504 20:52:03.256444 15684 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, next=2, gen=1, sticky=9223372036.854775807,2147483647]
I200504 20:52:03.313447 15684 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200504 20:52:03.333313 15684 kv/kvserver/store_snapshot.go:931  [n1,s1,r33/1:/{Table/53/1/6-Max}] sending LEARNER snapshot 6a13a6d8 at applied index 15
I200504 20:52:03.333449 16083 kv/kvserver/raft_snapshot_queue.go:126  [n1,raftsnapshot,s1,r33/1:/{Table/53/1/6-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200504 20:52:03.335095 15684 kv/kvserver/store_snapshot.go:974  [n1,s1,r33/1:/{Table/53/1/6-Max}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 11, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200504 20:52:03.350457 16052 kv/kvserver/replica_raftstorage.go:780  [n2,s2,r33/2:{-}] applying LEARNER snapshot [id=6a13a6d8 index=15]
I200504 20:52:03.364464 16052 kv/kvserver/replica_raftstorage.go:801  [n2,s2,r33/2:/{Table/53/1/6-Max}] applied LEARNER snapshot [total=14ms ingestion=4@11ms id=6a13a6d8 index=15]
I200504 20:52:03.377416 15684 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=2, sticky=9223372036.854775807,2147483647]
I200504 20:52:03.399292 15684 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200504 20:52:03.494643 16042 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1VOTER_DEMOTING]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]
I200504 20:52:03.556110 16042 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing ENTER_JOINT(r1 l1) REMOVE_REPLICA[(n1,s1):1VOTER_DEMOTING]: after=[(n1,s1):1VOTER_DEMOTING (n2,s2):2] next=3
I200504 20:52:03.565247 16042 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1VOTER_DEMOTING, (n2,s2):2, next=3, gen=4, sticky=9223372036.854775807,2147483647]
I200504 20:52:03.593066 16042 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing LEAVE_JOINT: after=[(n1,s1):1LEARNER (n2,s2):2] next=3
I200504 20:52:03.601888 16042 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1LEARNER]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1LEARNER, (n2,s2):2, next=3, gen=5, sticky=9223372036.854775807,2147483647]
I200504 20:52:03.658549 16042 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing SIMPLE(r1) REMOVE_REPLICA[(n1,s1):1LEARNER]: after=[(n2,s2):2] next=3
I200504 20:52:03.664579 14277 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/{Table/53/1/6-Max}] removing replica r33/1
I200504 20:52:03.744692 15684 sql/event_log.go:132  [n1,client=127.0.0.1:47614,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:sql.defaults.results_buffer.size Value:0 User:root}
I200504 20:52:03.873864 15005 gossip/gossip.go:1539  [n2] node has connected to cluster via gossip
I200504 20:52:03.875154 15005 kv/kvserver/stores.go:255  [n2] wrote 2 node addresses to persistent storage
--- FAIL: TestDrainingProcessorSwallowsUncertaintyError (3.72s)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true
    --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true (0.21s)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false
        --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false (0.07s)
            processors_test.go:663: ERROR: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1588625524.078933299,2 encountered previous write with future timestamp 1588625524.078933300,2 within uncertainty interval `t <= 1588625524.578933297,0`; observed timestamps: [{1 1588625524.078933297,0} {2 1588625524.092301335,0}] (SQLSTATE 40001)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestDrainingProcessorSwallowsUncertaintyError PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDrainingProcessorSwallowsUncertaintyError.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/rowexec).TestDrainingProcessorSwallowsUncertaintyError failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1920195&tab=buildLog) on [master@34344ebaf9a832e319ff8b65563490281acf3c0e](https://github.com/cockroachdb/cockroach/commits/34344ebaf9a832e319ff8b65563490281acf3c0e):

```
I200504 22:40:01.730119 14275 gossip/gossip.go:405  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:42589" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc3" > > ServerVersion:<major_val:20 minor_val:1 patch:0 unstable:2 > build_tag:"v20.1.0-beta.4-1294-g34344ebaf9" started_at:1588632001729455278 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:34471" > 
I200504 22:40:01.731200 14275 kv/kvserver/stores.go:236  [n3] read 0 node addresses from persistent storage
I200504 22:40:01.731901 14275 kv/kvserver/stores.go:255  [n3] wrote 2 node addresses to persistent storage
I200504 22:40:01.736370 14246 kv/kvserver/stores.go:255  [n1] wrote 2 node addresses to persistent storage
I200504 22:40:01.739231 15204 kv/kvserver/stores.go:255  [n2] wrote 2 node addresses to persistent storage
I200504 22:40:01.759190 14275 server/node.go:572  [n3] bootstrapped store [n3,s3]
I200504 22:40:01.762191 14275 server/node.go:472  [n3] node=3: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200504 22:40:01.763833 14275 server/server.go:1435  [n3] starting https server at 127.0.0.1:39559 (use: 127.0.0.1:39559)
I200504 22:40:01.763973 14275 server/server.go:1440  [n3] starting postgres server at 127.0.0.1:34471 (use: 127.0.0.1:34471)
I200504 22:40:01.764073 14275 server/server.go:1442  [n3] starting grpc server at 127.0.0.1:42589
I200504 22:40:01.764131 14275 server/server.go:1443  [n3] advertising CockroachDB node at 127.0.0.1:42589
I200504 22:40:01.770852 15720 server/server.go:1066  active cluster version is now 20.1-2 (up from 19.1-1)
I200504 22:40:01.865655 15769 sql/temporary_schema.go:453  [n3] skipping temporary object cleanup run as it is not the leaseholder
I200504 22:40:01.865849 15769 sql/temporary_schema.go:454  [n3] completed temporary object cleanup job
I200504 22:40:01.865959 15769 sql/temporary_schema.go:581  [n3] temporary object cleaner next scheduled to run at 2020-05-04 23:10:01.852272619 +0000 UTC
I200504 22:40:02.007040 14275 server/server_sql.go:636  [n3] done ensuring all necessary migrations have run
I200504 22:40:02.007289 14275 server/server.go:1753  [n3] serving sql connections
I200504 22:40:02.014480 15614 sqlmigrations/migrations.go:663  [n3] starting wait for upgrade finalization before schema change job migration
I200504 22:40:02.087970 15614 sqlmigrations/migrations.go:689  [n3] schema change job migration already complete
I200504 22:40:02.123598 15768 sql/event_log.go:132  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:127.0.0.1:42589 Attrs: Locality:region=test,dc=dc3 ServerVersion:20.1-2 BuildTag:v20.1.0-beta.4-1294-g34344ebaf9 StartedAt:1588632001729455278 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:34471} ClusterID:2a731782-ef44-42a7-abb4-e20369163896 StartedAt:1588632001729455278 LastUp:1588632001729455278}
I200504 22:40:02.172073 15767 server/server_update.go:55  [n3] no need to upgrade, cluster already at the newest version
I200504 22:40:02.206673 15828 sql/event_log.go:132  [n1,client=127.0.0.1:50062,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.range_merge.queue_enabled Value:false User:root}
I200504 22:40:02.343750 15828 sql/event_log.go:132  [n1,client=127.0.0.1:50062,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I200504 22:40:02.375436 15249 gossip/gossip.go:1539  [n2] node has connected to cluster via gossip
I200504 22:40:02.376871 15249 kv/kvserver/stores.go:255  [n2] wrote 2 node addresses to persistent storage
I200504 22:40:02.489063 15828 sql/event_log.go:132  [n1,client=127.0.0.1:50062,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:test.public.t Statement:CREATE TABLE test.public.t (x INT8 PRIMARY KEY) User:root}
I200504 22:40:02.604066 15828 kv/kvserver/replica_command.go:397  [n1,s1,r32/1:/{Table/36-Max}] initiating a split of this range at key /Table/53/1/6 [r33] (manual)
I200504 22:40:02.670734 15828 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, next=2, gen=1, sticky=9223372036.854775807,2147483647]
I200504 22:40:02.709553 15828 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200504 22:40:02.737433 15828 kv/kvserver/store_snapshot.go:931  [n1,s1,r33/1:/{Table/53/1/6-Max}] sending LEARNER snapshot 65657dbf at applied index 15
I200504 22:40:02.737831 15904 kv/kvserver/raft_snapshot_queue.go:126  [n1,raftsnapshot,s1,r33/1:/{Table/53/1/6-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200504 22:40:02.739106 15828 kv/kvserver/store_snapshot.go:974  [n1,s1,r33/1:/{Table/53/1/6-Max}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 11, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200504 22:40:02.745592 16129 kv/kvserver/replica_raftstorage.go:780  [n2,s2,r33/2:{-}] applying LEARNER snapshot [id=65657dbf index=15]
I200504 22:40:02.762977 16129 kv/kvserver/replica_raftstorage.go:801  [n2,s2,r33/2:/{Table/53/1/6-Max}] applied LEARNER snapshot [total=17ms ingestion=4@13ms id=65657dbf index=15]
I200504 22:40:02.772207 15828 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=2, sticky=9223372036.854775807,2147483647]
I200504 22:40:02.799896 15828 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200504 22:40:02.908259 16149 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1VOTER_DEMOTING]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]
I200504 22:40:02.986029 16149 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing ENTER_JOINT(r1 l1) REMOVE_REPLICA[(n1,s1):1VOTER_DEMOTING]: after=[(n1,s1):1VOTER_DEMOTING (n2,s2):2] next=3
I200504 22:40:02.996302 16149 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1VOTER_DEMOTING, (n2,s2):2, next=3, gen=4, sticky=9223372036.854775807,2147483647]
I200504 22:40:03.027006 16149 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing LEAVE_JOINT: after=[(n1,s1):1LEARNER (n2,s2):2] next=3
I200504 22:40:03.038427 16149 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1LEARNER]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1LEARNER, (n2,s2):2, next=3, gen=5, sticky=9223372036.854775807,2147483647]
I200504 22:40:03.095787 16149 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing SIMPLE(r1) REMOVE_REPLICA[(n1,s1):1LEARNER]: after=[(n2,s2):2] next=3
I200504 22:40:03.103303 14377 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/{Table/53/1/6-Max}] removing replica r33/1
I200504 22:40:03.171268 15828 sql/event_log.go:132  [n1,client=127.0.0.1:50062,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:sql.defaults.results_buffer.size Value:0 User:root}
--- FAIL: TestDrainingProcessorSwallowsUncertaintyError (4.62s)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false
        --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false (0.03s)
            processors_test.go:663: ERROR: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1588632003.573366535,0 encountered previous write with future timestamp 1588632003.573366536,0 within uncertainty interval `t <= 1588632004.073366535,0`; observed timestamps: [{1 1588632003.573366535,0}] (SQLSTATE 40001)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true
    --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true (0.19s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestDrainingProcessorSwallowsUncertaintyError PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDrainingProcessorSwallowsUncertaintyError.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/rowexec).TestDrainingProcessorSwallowsUncertaintyError failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1921266&tab=buildLog) on [master@425eaa8fb05fc32b2c42827b85338daa52f4177c](https://github.com/cockroachdb/cockroach/commits/425eaa8fb05fc32b2c42827b85338daa52f4177c):

```
I200505 06:22:53.850018 15005 gossip/server.go:227  [n1] received initial cluster-verification connection from 127.0.0.1:42639
I200505 06:22:53.919662 13897 server/node.go:369  [n?] new node allocated ID 3
I200505 06:22:53.919993 13897 gossip/gossip.go:405  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:42639" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc3" > > ServerVersion:<major_val:20 minor_val:1 patch:0 unstable:2 > build_tag:"v20.1.0-beta.4-1296-g425eaa8fb0" started_at:1588659773919743423 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:37677" > 
I200505 06:22:53.920875 13897 kv/kvserver/stores.go:236  [n3] read 0 node addresses from persistent storage
I200505 06:22:53.921326 13897 kv/kvserver/stores.go:255  [n3] wrote 2 node addresses to persistent storage
I200505 06:22:53.922869 13900 kv/kvserver/stores.go:255  [n1] wrote 2 node addresses to persistent storage
I200505 06:22:53.924593 14605 kv/kvserver/stores.go:255  [n2] wrote 2 node addresses to persistent storage
I200505 06:22:53.941027 13897 server/node.go:572  [n3] bootstrapped store [n3,s3]
I200505 06:22:53.941973 13897 server/node.go:472  [n3] node=3: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200505 06:22:53.943165 15274 server/server.go:1066  active cluster version is now 20.1-2 (up from 19.1-1)
I200505 06:22:53.943404 13897 server/server.go:1435  [n3] starting https server at 127.0.0.1:43165 (use: 127.0.0.1:43165)
I200505 06:22:53.943598 13897 server/server.go:1440  [n3] starting postgres server at 127.0.0.1:37677 (use: 127.0.0.1:37677)
I200505 06:22:53.943672 13897 server/server.go:1442  [n3] starting grpc server at 127.0.0.1:42639
I200505 06:22:53.943738 13897 server/server.go:1443  [n3] advertising CockroachDB node at 127.0.0.1:42639
I200505 06:22:54.039197 15311 sql/temporary_schema.go:453  [n3] skipping temporary object cleanup run as it is not the leaseholder
I200505 06:22:54.039330 15311 sql/temporary_schema.go:454  [n3] completed temporary object cleanup job
I200505 06:22:54.039452 15311 sql/temporary_schema.go:581  [n3] temporary object cleaner next scheduled to run at 2020-05-05 06:52:54.038785546 +0000 UTC
I200505 06:22:54.046584 13897 server/server_sql.go:636  [n3] done ensuring all necessary migrations have run
I200505 06:22:54.046733 13897 server/server.go:1753  [n3] serving sql connections
I200505 06:22:54.048354 15152 sqlmigrations/migrations.go:663  [n3] starting wait for upgrade finalization before schema change job migration
I200505 06:22:54.064395 15152 sqlmigrations/migrations.go:689  [n3] schema change job migration already complete
I200505 06:22:54.097954 15310 sql/event_log.go:132  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:127.0.0.1:42639 Attrs: Locality:region=test,dc=dc3 ServerVersion:20.1-2 BuildTag:v20.1.0-beta.4-1296-g425eaa8fb0 StartedAt:1588659773919743423 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:37677} ClusterID:18433b2a-30f0-40ff-b58b-84f58c280a59 StartedAt:1588659773919743423 LastUp:1588659773919743423}
I200505 06:22:54.116244 15309 server/server_update.go:55  [n3] no need to upgrade, cluster already at the newest version
I200505 06:22:54.135781 15412 sql/event_log.go:132  [n1,client=127.0.0.1:46596,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.range_merge.queue_enabled Value:false User:root}
I200505 06:22:54.193156 15412 sql/event_log.go:132  [n1,client=127.0.0.1:46596,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I200505 06:22:54.239054 15412 sql/event_log.go:132  [n1,client=127.0.0.1:46596,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:test.public.t Statement:CREATE TABLE test.public.t (x INT8 PRIMARY KEY) User:root}
I200505 06:22:54.292691 15412 kv/kvserver/replica_command.go:397  [n1,s1,r32/1:/{Table/36-Max}] initiating a split of this range at key /Table/53/1/6 [r33] (manual)
I200505 06:22:54.331907 15412 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, next=2, gen=1, sticky=9223372036.854775807,2147483647]
I200505 06:22:54.345531 15412 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200505 06:22:54.352039 15412 kv/kvserver/store_snapshot.go:931  [n1,s1,r33/1:/{Table/53/1/6-Max}] sending LEARNER snapshot eb030aaa at applied index 14
I200505 06:22:54.352747 15412 kv/kvserver/store_snapshot.go:974  [n1,s1,r33/1:/{Table/53/1/6-Max}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 11, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200505 06:22:54.354772 15686 kv/kvserver/raft_snapshot_queue.go:126  [n1,raftsnapshot,s1,r33/1:/{Table/53/1/6-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200505 06:22:54.355391 15640 kv/kvserver/replica_raftstorage.go:780  [n2,s2,r33/2:{-}] applying LEARNER snapshot [id=eb030aaa index=14]
I200505 06:22:54.358657 15640 kv/kvserver/replica_raftstorage.go:801  [n2,s2,r33/2:/{Table/53/1/6-Max}] applied LEARNER snapshot [total=3ms ingestion=4@2ms id=eb030aaa index=14]
I200505 06:22:54.361945 15412 kv/kvserver/replica_command.go:1602  [n1,s1,r33/1:/{Table/53/1/6-Max}] change replicas (add [(n2,s2):2] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=2, sticky=9223372036.854775807,2147483647]
I200505 06:22:54.377667 15412 kv/kvserver/replica_raft.go:247  [n1,s1,r33/1:/{Table/53/1/6-Max}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200505 06:22:54.448249 15507 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1VOTER_DEMOTING]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]
I200505 06:22:54.480656 15507 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing ENTER_JOINT(r1 l1) REMOVE_REPLICA[(n1,s1):1VOTER_DEMOTING]: after=[(n1,s1):1VOTER_DEMOTING (n2,s2):2] next=3
I200505 06:22:54.486580 15507 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove []): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1VOTER_DEMOTING, (n2,s2):2, next=3, gen=4, sticky=9223372036.854775807,2147483647]
I200505 06:22:54.499007 15507 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing LEAVE_JOINT: after=[(n1,s1):1LEARNER (n2,s2):2] next=3
I200505 06:22:54.504530 15507 kv/kvserver/replica_command.go:1602  [n2,s2,r33/2:/{Table/53/1/6-Max}] change replicas (add [] remove [(n1,s1):1LEARNER]): existing descriptor r33:/{Table/53/1/6-Max} [(n1,s1):1LEARNER, (n2,s2):2, next=3, gen=5, sticky=9223372036.854775807,2147483647]
I200505 06:22:54.536369 15507 kv/kvserver/replica_raft.go:247  [n2,s2,r33/2:/{Table/53/1/6-Max}] proposing SIMPLE(r1) REMOVE_REPLICA[(n1,s1):1LEARNER]: after=[(n2,s2):2] next=3
I200505 06:22:54.539154 14014 kv/kvserver/store_remove_replica.go:122  [n1,s1,r33/1:/{Table/53/1/6-Max}] removing replica r33/1
I200505 06:22:54.571140 15412 sql/event_log.go:132  [n1,client=127.0.0.1:46596,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:sql.defaults.results_buffer.size Value:0 User:root}
--- FAIL: TestDrainingProcessorSwallowsUncertaintyError (2.36s)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false
        --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true/dummy=false (0.02s)
            processors_test.go:663: ERROR: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1588659774.775772202,2 encountered previous write with future timestamp 1588659774.775772203,2 within uncertainty interval `t <= 1588659775.275772200,0`; observed timestamps: [{1 1588659774.775772200,0} {2 1588659774.779175650,0}] (SQLSTATE 40001)
=== RUN   TestDrainingProcessorSwallowsUncertaintyError/vectorize=true
    --- FAIL: TestDrainingProcessorSwallowsUncertaintyError/vectorize=true (0.12s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestDrainingProcessorSwallowsUncertaintyError PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDrainingProcessorSwallowsUncertaintyError.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/rowexec/processors_test.go
+++ b/pkg/sql/rowexec/processors_test.go
@@ -606,23 +606,31 @@ func TestDrainingProcessorSwallowsUncertaintyError(t *testing.T) {
 
 	atomic.StoreInt64(&trapRead, 1)
 
-	// Run with the default vectorize mode and the explicit "on" mode.
+	// Run with the vectorize off and on.
 	testutils.RunTrueAndFalse(t, "vectorize", func(t *testing.T, vectorize bool) {
 		// We're going to run the test twice in each vectorize configuration. Once
 		// in "dummy" node, which just verifies that the test is not fooling itself
 		// by increasing the limit from 5 to 6 and checking that we get the injected
 		// error in that case.
 		testutils.RunTrueAndFalse(t, "dummy", func(t *testing.T, dummy bool) {
+			// Reset the blocking condition.
+			blockedRead.Lock()
+			blockedRead.shouldUnblock = false
+			blockedRead.Unlock()
 			// Force DistSQL to distribute the query. Otherwise, as of Nov 2018, it's hard
 			// to convince it to distribute a query that uses an index.
 			if _, err := conn.Exec("set distsql='always'"); err != nil {
 				t.Fatal(err)
 			}
+			vectorizeMode := "off"
 			if vectorize {
-				if _, err := conn.Exec("set vectorize='on'"); err != nil {
-					t.Fatal(err)
-				}
+				vectorizeMode = "on"
 			}
+
+			if _, err := conn.Exec(fmt.Sprintf("set vectorize='%s'; set vectorize_row_count_threshold=0", vectorizeMode)); err != nil {
+				t.Fatal(err)
+			}
+
 			limit := 5
 			if dummy {
 				limit = 6

